data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeOperationStatus.avsc

```diff
@@ -43,28 +31,23 @@
     },
     {
       "name": "heartbeatTimestamp",
-      "type": {
-        "type": "long",
-        "logicalType": "timestamp-millis"
-      },
+      "type": [
+        "null",
+        {
+          "type": "long",
+          "logicalType": "timestamp-millis"
+        }
+      ],
```

Comment:
At the moment, upgrading a CPI is generating an Operation Status with a null heartbeatTimestamp. I've updated the schema to allow this, but maybe that's something we need to change in the implementation of the Update instead, so just wanted to highlight this.

Reply:
Good point, yes we can just make this nullable as it's nullable in the database too